lib/app/features/config/providers/config_provider.c.dart

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:io';
+
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/core/providers/dio_provider.c.dart';
+import 'package:ion/app/features/core/providers/env_provider.c.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'config_provider.c.g.dart';
+
+@riverpod
+class ConfigNotifier extends _$ConfigNotifier {
```

Comment:
Do we need this `ConfigNotifier`? You could only change a method `fetchConfigForCurrentPlatform` into a functional provider